lib/widgets/content.dart

```diff
@@ -941,7 +953,80 @@ class _KatexSpan extends StatelessWidget {
         textAlign: textAlign,
         child: widget);
     }
-    return widget;
+
+    final marginRight = switch (styles.marginRightEm) {
+      double marginRightEm => marginRightEm * em,
+      null => null,
+    };
+    final marginLeft = switch (styles.marginLeftEm) {
+      double marginLeftEm => marginLeftEm * em,
+      null => null,
+    };
+
+    EdgeInsets? margin;
+    if (marginRight != null || marginLeft != null) {
+      margin = EdgeInsets.zero;
+      if (marginRight != null) {
+        assert(marginRight >= 0);
+        margin += EdgeInsets.only(right: marginRight);
+      }
+      if (marginLeft != null) {
+        assert(marginLeft >= 0);
+        margin += EdgeInsets.only(left: marginLeft);
+      }
+    }
+
+    return Container(
+      margin: margin,
+      height: styles.heightEm != null
+        ? styles.heightEm! * em
+        : null,
+      transform: styles.topEm != null
+        ? Matrix4.translationValues(0, styles.topEm! * em, 0)
+        : null,
+      child: widget,
+    );
+  }
+}
+
+class _KatexStrut extends StatelessWidget {
+  const _KatexStrut(this.node);
+
+  final KatexStrutNode node;
+
+  @override
+  Widget build(BuildContext context) {
+    final em = DefaultTextStyle.of(context).style.fontSize!;
+
+    final verticalAlignEm = node.verticalAlignEm;
+    if (verticalAlignEm == null) {
+      return SizedBox(height: node.heightEm * em);
+    }
+
+    return SizedBox(
+      height: node.heightEm * em,
+      child: Baseline(
+        baseline: (verticalAlignEm + node.heightEm) * em,
+        baselineType: TextBaseline.alphabetic,
+        child: const Text('')),
```

Comment:
Can we write a test for this?

What's an example of an expression that would look wrong if we left out this Baseline widget; and what would be the effect on how it looks?

Reply:
One way to do it that works before this commit and remains clean after it might be with `example` and `characters` as positional items in the record but `skip` named:
```dart
      const testCases = <(ContentExample, List<(String, Offset, Size)>, {bool? skip})>[
        (ContentExample.mathBlockKatexSizing, skip: false, [
          ('1', Offset(0.00, 2.24), Size(25.59, 61.00)),
          // …
```

Then this commit would change a `skip: true` to `skip: false`, and perhaps change some of the expected numbers.

Reply:
Updated.